packages/nodes-base/nodes/Google/Docs/GoogleDocs.node.ts

```diff
@@ -0,0 +1,415 @@
+import {
+	IExecuteFunctions,
+} from 'n8n-core';
+
+import {
+	IDataObject,
+	INodeExecutionData,
+	INodeType,
+	INodeTypeDescription,
+} from 'n8n-workflow';
+
+import {
+	googleApiRequest,
+} from './GenericFunctions';
+
+import {
+	documentFields,
+	documentOperations,
+} from './DocumentDescription';
+
+
+export class GoogleDocs implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'Google Docs',
+		name: 'googleDocs',
+		icon: 'file:googleDocs.svg',
+		group: ['input'],
+		version: 1,
+		subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
+		description: 'Consume Google Docs API.',
+		defaults: {
+			name: 'Google Docs',
+			color: '#1a73e8',
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		credentials: [
+			{
+				name: 'googleDocsOAuth2Api',
+				required: true,
+			},
+		],
+		properties: [
+			{
+				displayName: 'Resource',
+				name: 'resource',
+				type: 'options',
+				options: [
+					{
+						name: 'Document',
+						value: 'document',
+					},
+				],
+				default: 'document',
+				description: 'The resource to operate on.',
+			},
+			...documentOperations,
+			...documentFields,
+		],
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: IDataObject[] = [];
+		const length = items.length;
+		const qs: IDataObject = {};
+		let responseData;
+		const resource = this.getNodeParameter('resource', 0) as string;
+		const operation = this.getNodeParameter('operation', 0) as string;
+		for (let i = 0; i < length; i++) {
+			try {
+				if (resource === 'document') {
+					//https://developers.google.com/docs/api/reference/rest/v1/documents/create
+					if (operation === 'create') {
+						const title = this.getNodeParameter('title', i) as string;
+
+						const body: IDataObject = {
+							title,
+						};
+
+						responseData = await googleApiRequest.call(
+							this,
+							'POST',
+							'/documents',
+							body,
+							qs,
+						);
+
+						console.log({operation,responseData});
+
+					}
+					//https://developers.google.com/docs/api/reference/rest/v1/documents/get
+					if (operation === 'get') {
+						const documentId = this.getNodeParameter('documentId', i) as string;
+
+						responseData = await googleApiRequest.call(
+							this,
+							'GET',
+							`/documents/${documentId}`,
+							{},
+							qs,
+						);
+
+					}
+					//https://developers.google.com/docs/api/reference/rest/v1/documents/batchUpdate
+					if (operation === 'update') {
+
+						const documentId = this.getNodeParameter('documentId', i) as string;
+
+						const updateFields = this.getNodeParameter('updateFields', i) as { writeControl?: IDataObject, requestsUi?: IDataObject };
+
+						const body: IDataObject = {
+							documentId,
+							requests: [],
+						};
```

Comment:
The idea is to type your objects once at full:

```ts
const updateFields = this.getNodeParameter('updateFields', i) as {
	writeControl?: { control: string; writeControlObject: { control: string, value: string }  },
	requestsUi?: IDataObject
};

const body = {
	documentId,
	requests: [],
} as { documentId: string; requests: IDataObject[]; writeControl?: IDataObject };
```

So that you can clear out all of the castings from your logic below:

```ts
if (updateFields.writeControl) {
	const writeControl = updateFields.writeControl.writeControlObject;
	body.writeControl = {
		[writeControl.control]: writeControl.value,
	};
}
```

This is just an example to build on. Ideally not inline, but extracted out into a type.